athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/ElasticsearchRecordHandler.java

```diff
@@ -0,0 +1,215 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.connectors.athena.elasticsearch;
+
+import com.amazonaws.athena.connector.lambda.QueryStatusChecker;
+import com.amazonaws.athena.connector.lambda.data.Block;
+import com.amazonaws.athena.connector.lambda.data.BlockSpiller;
+import com.amazonaws.athena.connector.lambda.handlers.RecordHandler;
+import com.amazonaws.athena.connector.lambda.records.ReadRecordsRequest;
+import com.amazonaws.services.athena.AmazonAthena;
+import com.amazonaws.services.athena.AmazonAthenaClientBuilder;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.secretsmanager.AWSSecretsManager;
+import com.amazonaws.services.secretsmanager.AWSSecretsManagerClientBuilder;
+import org.apache.arrow.util.VisibleForTesting;
+import org.apache.arrow.vector.types.Types;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static com.amazonaws.connectors.athena.elasticsearch.ElasticsearchFieldResolver.DEFAULT_FIELD_RESOLVER;
+
+/**
+ * This class is part of an tutorial that will walk you through how to build a connector for your
+ * custom data source. The README for this module (athena-elasticsearch) will guide you through preparing
+ * your development environment, modifying this elasticsearch RecordHandler, building, deploying, and then
+ * using your new source in an Athena query.
+ * <p>
+ * More specifically, this class is responsible for providing Athena with actual rows level data from your source. Athena
+ * will call readWithConstraint(...) on this class for each 'Split' you generated in ElasticsearchMetadataHandler.
+ * <p>
+ * For more elasticsearchs, please see the other connectors in this repository (e.g. athena-cloudwatch, athena-docdb, etc...)
+ */
+public class ElasticsearchRecordHandler
+        extends RecordHandler
+{
+    private static final Logger logger = LoggerFactory.getLogger(ElasticsearchRecordHandler.class);
+
+    /**
+     * used to aid in debugging. Athena will use this name in conjunction with your catalog id
+     * to correlate relevant query errors.
+     */
+    private static final String SOURCE_TYPE = "elasticsearch";
+
+    // Pagination batch size (100 documents).
+    private static final int QUERY_BATCH_SIZE = 100;
```

Comment:
is 100 the right number? how did we come up with it?

Reply:
I'm using the same value that the docdb connector is using. I'm open to suggestions regarding the appropriate pagination batch size.